frontend/src/services/settings.ts

```diff
@@ -5,16 +5,22 @@ export type Settings = {
   AGENT: string;
   LANGUAGE: string;
   LLM_API_KEY: string;
+  CONFIRMATION_MODE: boolean;
 };
 
+export type SettingsInput = {
+  [K in keyof Settings]: Settings[K];
+}[keyof Settings];
+
 export const DEFAULT_SETTINGS: Settings = {
   LLM_MODEL: "gpt-4o",
   AGENT: "CodeActAgent",
   LANGUAGE: "en",
   LLM_API_KEY: "",
+  CONFIRMATION_MODE: false,
 };
 
-const validKeys = Object.keys(DEFAULT_SETTINGS) as (keyof Settings)[];
+const validKeys = Object.keys(DEFAULT_SETTINGS) as (keyof SettingsInput)[];
```

Comment:
This is an incorrect change, types such as `string | boolean` do not have keys, `validKeys` end up with type `"valueOf"[]` (inferred from the common methods between `String.prototype` and `Boolean.prototype`)

```suggestion
const validKeys = Object.keys(DEFAULT_SETTINGS) as (keyof Settings)[];
```
The code now correctly infers the types of `validKeys` to be `(keyof Settings)[]`